website/docs/index.html.markdown

```diff
@@ -33,33 +33,17 @@ resource "github_membership" "membership_for_user_x" {
 
 The following arguments are supported in the `provider` block:
 
-* `token` - (Optional) This is the GitHub personal access token. It can also be
-  sourced from the `GITHUB_TOKEN` environment variable. If `anonymous` is false,
-  `token` is required.
+* `token` - (Required) This is the GitHub personal access token. It must be provided, but
+  it can also be sourced from the `GITHUB_TOKEN` environment variable.
 
-* `organization` - (Optional) This is the target GitHub organization to manage.
-  The account corresponding to the token will need "owner" privileges for this
-  organization. It can also be sourced from the `GITHUB_ORGANIZATION`
-  environment variable. If `individual` is set to `false`, organization is required.
+* `owner` - (Optional) This is the target GitHub organization or a user to manage. The account
+  corresponding to the token will need "owner" privileges for this organization. It must be provided, but
+  it can also be sourced from the `GITHUB_OWNER` environment variable.
+
+* `organization` - (DEPRECATED) This is the target GitHub organization or a user to manage. The account
+  corresponding to the token will need "organization" privileges for this organization. It must be provided, but
+  it can also be sourced from the `GITHUB_ORGANIZATION` environment variable.
 
 * `base_url` - (Optional) This is the target GitHub base API endpoint. Providing a value is a
   requirement when working with GitHub Enterprise.  It is optional to provide this value and
   it can also be sourced from the `GITHUB_BASE_URL` environment variable.  The value must end with a slash.
-  `https://github.someorg.example/api/`.
-
-* `insecure` - (Optional) Whether server should be accessed without verifying the TLS certificate.
-  As the name suggests **this is insecure** and should not be used beyond experiments,
-  accessing local (non-production) GHE instance etc.
-  There is a number of ways to obtain trusted certificate for free, e.g. from [Let's Encrypt](https://letsencrypt.org/).
-  Such trusted certificate *does not require* this option to be enabled.
-  Defaults to `false`.
-
-* `individual`: (Optional) Run outside an organization.  When `individual` is true, the provider will run outside
-  the scope of an organization. Defaults to `false`.
-
-* `anonymous`: (Optional) Authenticate without a token.  When `anonymous` is true, the provider will not be able to
```

Comment:
We should call out this removal explicitly when we update the CHANGELOG for the release this goes out in.

Reply:
why this was removed and not added to the docs ?, I had to come here to actually find out why

Reply:
Damn, yes, it took me 2 hours to find out, and now I'm hardcoding the old plugin version. I need anonymous functionality to just pull the GitHub API IPs. That should never require tokens.

Reply:
The CHANGELOG has been retroactively updated to help save time for anyone else affected.  

Thanks for being quick to call this out.

Reply:
@MarcoDalco open a bug report to make sure such kind of requests are possible? The advantage of token is that it probably raises from GitHub API limits, but I haven't checked. A list of resources that one can use without token would be handy.

Reply:
@abitrolly https://api.github.com/meta?q=hooks "You can add the IP address we use when delivering hooks to your server's allow list. To ensure that you're always checking the right IP address, you can use the /meta endpoint to find the address we use." https://developer.github.com/v3/guides/best-practices-for-integrators/ . Having to periodically update the whitelisted IPs is not great, but it's what seems to be suggested, at the moment. Maybe in the future we could discriminate on some content/token, but right now whitelisting seems to be done that way, based on IPs.